server/tests/api_v4/authenticated/test_user_update.py

```diff
@@ -33,3 +42,212 @@ async def test_authenticated_user_update_ok(coolorg: CoolorgRpcClients, backend:
+        now=t1,
+        organization_id=coolorg.organization_id,
+        author=coolorg.alice.device_id,
+        author_verify_key=coolorg.alice.signing_key.verify_key,
+        revoked_user_certificate=certif1.dump_and_sign(coolorg.alice.signing_key),
+    )
+
+    assert isinstance(outcome, RevokedUserCertificate)
+
+    # 2) Try to update user Bob (which is now revoked)
+
+    now = DateTime.now()
+    certif = UserUpdateCertificate(
+        author=coolorg.alice.device_id,
+        timestamp=now,
+        user_id=coolorg.bob.device_id.user_id,
+        new_profile=UserProfile.ADMIN,
+    )
+
+    rep = await coolorg.alice.user_update(
+        user_update_certificate=certif.dump_and_sign(coolorg.alice.signing_key)
+    )
+    assert rep == authenticated_cmds.v4.user_update.RepUserRevoked()
+
+
+async def test_authenticated_user_update_user_no_changes(
+    coolorg: CoolorgRpcClients, backend: Backend
+) -> None:
+    now = DateTime.now()
+    certif = UserUpdateCertificate(
+        author=coolorg.alice.device_id,
+        timestamp=now,
+        user_id=coolorg.bob.device_id.user_id,
+        new_profile=UserProfile.STANDARD,
+    )
+
+    rep = await coolorg.alice.user_update(
+        user_update_certificate=certif.dump_and_sign(coolorg.alice.signing_key)
+    )
+    assert rep == authenticated_cmds.v4.user_update.RepUserNoChanges()
+
+
+@pytest.mark.parametrize(
+    "kind",
+    (
+        "user_update_certificate",
+        "user_update_certificate_not_author_user_id",
+        "user_update_certificate_author_device_mismatch",
+        "user_update_certificate_author_update_itself",
+    ),
+)
+async def test_authenticated_user_update_invalid_certificate(
+    coolorg: CoolorgRpcClients, backend: Backend, kind: str
+) -> None:
+    now = DateTime.now()
+
+    match kind:
+        case "user_update_certificate":
+            certif = b"<dummy>"
+        case "user_update_certificate_not_author_user_id":
+            certif = UserUpdateCertificate(
+                author=coolorg.bob.device_id,
+                timestamp=now,
+                user_id=coolorg.bob.device_id.user_id,
+                new_profile=UserProfile.ADMIN,
+            ).dump_and_sign(coolorg.alice.signing_key)
+        case "user_update_certificate_author_device_mismatch":
+            certif = UserUpdateCertificate(
+                author=DeviceID("alice@dev2"),
+                timestamp=now,
+                user_id=coolorg.bob.device_id.user_id,
+                new_profile=UserProfile.ADMIN,
+            ).dump_and_sign(coolorg.alice.signing_key)
+        case "user_update_certificate_author_update_itself":
+            certif = UserUpdateCertificate(
+                author=coolorg.alice.device_id,
+                timestamp=now,
+                user_id=coolorg.alice.device_id.user_id,
+                new_profile=UserProfile.STANDARD,
+            ).dump_and_sign(coolorg.alice.signing_key)
+        case unknown:
+            assert False, unknown
```

Comment:
Instead of a match, you could use a fixture for each action kind, and request the result with `certif = request.getfixture(kind)`.

I'm not sure if that would be better :thinking:

Reply:
I'm not sure either. I think is important to be consistent with the rest of the test code base (which uses `match` on a `kind` paramter).